src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AddMissingInterpolationToken.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeQuality.Analyzers.QualityGuidelines
+{
+    using static MicrosoftCodeQualityAnalyzersResources;
+    public abstract class AbstractAddMissingInterpolationTokenAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA2259";
+
+        internal static DiagnosticDescriptor Rule = DiagnosticDescriptorHelper.Create(RuleId,
+            CreateLocalizableResourceString(nameof(AddMissingInterpolationTokenTitle)),
+            CreateLocalizableResourceString(nameof(AddMissingInterpolationTokenMessage)),
+            DiagnosticCategory.Usage,
+            RuleLevel.IdeSuggestion,
```

Comment:
@Evangelink I'm not sure. @stephentoub What do you think? (btw, I'm not sure the rule requires review in dotnet/runtime first)

Reply:
> Is it the right level?

Yes, I think suggestion makes sense as the default level, assuming we're confident this will have a very low false positive rate.  We don't want to add lots of bugs if someone does a fix-all.

> I'm not sure the rule requires review in dotnet/runtime first

It doesn't.  This is about C# syntax / misuse rather than about APIs.